common/djangoapps/cache_toolbox/middleware.py

```diff
@@ -106,6 +109,19 @@ def process_request(self, request):
                 )
                 # Raise an exception to fall through to the except clause below.
                 raise Exception
+            # Ensure that the user's session hash hasn't changed. We
+            # check that SessionAuthenticationMiddleware is enabled in
+            # order to match Django's behavior.
+            session_auth_class = 'django.contrib.auth.middleware.SessionAuthenticationMiddleware'
+            session_auth_enabled = session_auth_class in settings.MIDDLEWARE_CLASSES
+            auto_auth_enabled = settings.FEATURES.get('AUTOMATIC_AUTH_FOR_TESTING', False)
+            if not auto_auth_enabled and session_auth_enabled and hasattr(request.user, 'get_session_auth_hash'):
```

Comment:
Hmm.. come to think of it, why do you need to special case auto-authed users?  If their password is changed, they should still be logged out of their other sessions.